packages/components/focus-trap/src/focus-trap.vue

```diff
@@ -175,7 +175,11 @@ export default defineComponent({
           // Give embedded focus layer time to pause this layer before reclaiming focus
           // And only reclaim focus if it should currently be trapping
           setTimeout(() => {
-            if (!focusLayer.paused && props.trapped) {
+            if (
+              !focusLayer.paused &&
+              props.trapped &&
+              trapContainer.contains(document.activeElement)
```

Comment:
So, the intent of this design was to allow the consuming component of the focus trap to control whether or not focus should be trapped by using `props.trapped`. This code is basically negating that functionality. You could definitely end up breaking a ton of other components with this.

If you want the trap to auto-escape on tab, do something like this:

```
<template>
    <focus-trap :trapped="isTrapped" @focusout-prevented="onFocusOutTrap" />
</template>
<script>
const isTrapped = ref(false);

function onFocusOutTrap() {
    isTrapped.value = false;
}
</script>

```

Reply:
Alright

Reply:
This is the problematic line:

https://github.com/opengraphica/element-plus/blob/dev/packages/components/time-picker/src/common/picker.vue#L370

It was previously `pickerVisible.value = true`

`e?.relatedTarget` isn't going to exist the first time you click on the picker. I'm trying to find when that change was made to see why it was done.